File: apps/myjobhunter/backend/app/schemas/company/company.py
"""Company schemas — Phase 1 stub."""
import uuid
from datetime import datetime

from pydantic import BaseModel


class CompanyRead(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    name: str
    primary_domain: str | None
    industry: str | None
    size_range: str | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: apps/myjobhunter/frontend/src/types/company.ts
/**
 * TypeScript model for a Company as returned by the MJH backend.
 * Mirrors `CompanyResponse` in
 * apps/myjobhunter/backend/app/schemas/company/company_response.py.
 */
export interface Company {
  id: string;
  user_id: string;
  name: string;
  primary_domain: string | null;
  logo_url: string | null;
  industry: string | null;
  size_range: string | null;
  hq_location: string | null;
  description: string | null;
  external_ref: string | null;
  external_source: string | null;
  crunchbase_id: string | null;
  created_at: string;
  updated_at: string;
}
